modules/standard/GPU.chpl

```diff
@@ -189,8 +190,15 @@ module GPU
     :arg size: the number of elements in each GPU thread block's copy of the array.
    */
   inline proc createSharedArray(type eltType, param size): c_ptr(eltType) {
-    const voidPtr = __primitive("gpu allocShared", numBytes(eltType)*size);
-    return voidPtr : c_ptr(eltType);
+    if CHPL_GPU != "cpu" {
+      const voidPtr = __primitive("gpu allocShared", numBytes(eltType)*size);
+      return voidPtr : c_ptr(eltType);
+    }
+    else {
```

Comment:
Any reason not to do this in the runtime instead? `__primitive("gpu allocShared")` just calls out to a runtime function that we have different `impl` functions for. Couldn't you just have that call `chpl_malloc` or whatever?

I don't know if it's always possible, but I think it's preferable if we can get as much of our vendor-specific code into the vendor specific runtime libs. It also has the added benefit that things would still work if the primitive (for whatever reason) was called directly (either explicitly in Chapel code or if we code gen'd it somewhere else for whatever reason).

Reply:
> `__primitive("gpu allocShared")` just calls out to a runtime function that we have different impl functions for.

Does it? It doesn't look like it https://github.com/chapel-lang/chapel/blob/main/compiler/codegen/cg-expr.cpp#L5268

I think that should be doable. If that was the case, I agree with your sentiment here. But I am not willing to add that to this PR.

Reply:
Ah you're right, I must have been mixing this up with what we do for sync threads. Ok, I'm fine with how it is in your PR.